{tools}[foss/2025b] PyTorch v2.9.1, parameterized v0.9.0, pytest-subtests v0.15.0, ... w/ CUDA 12.9.1

Flamefire opened the pull request:
(created using `eb --new-pr`)

Flamefire committed:
adding easyconfigs: parameterized-0.9.0-GCCcore-14.3.0.eb, pytest-subtests-0.15.0-GCCcore-14.3.0.eb, PyTorch-2.9.1-foss-2025b-CUDA-12.9.1.eb, unittest-xml-reporting-3.2.0-GCCcore-14.3.0.eb and patches: PyTorch-1.12.1_add-hypothesis-suppression.patch, PyTorch-1.7.0_disable-dev-shm-test.patch, PyTorch-2.0.1_skip-tests-skipped-in-subprocess.patch, PyTorch-2.1.0_remove-test-requiring-online-access.patch, PyTorch-2.6.0_show-test-duration.patch, PyTorch-2.6.0_skip-test_segfault.patch, PyTorch-2.7.0_avoid_caffe2_test_cpp_jit.patch, PyTorch-2.7.1_avoid-caffe2-sandcastle-test-lib.patch, PyTorch-2.7.1_skip-test_data_parallel_rnn.patch, PyTorch-2.7.1_skip-test_gds_fails_in_ci.patch, PyTorch-2.7.1_skip-test_mixed_mm_exhaustive_dtypes.patch, PyTorch-2.7.1_skip-tests-requiring-SM90.patch, PyTorch-2.7.1_suport-64bit-BARs.patch, PyTorch-2.7.1_tolerance-test_partial_flat_weights.patch, PyTorch-2.9.0_disable-test_nan_assert.patch, PyTorch-2.9.0_enable-symbolizer-in-test_workspace_allocation_error.patch, PyTorch-2.9.0_fix-attention-squeeze.patch, PyTorch-2.9.0_fix-FP16-CPU-tests-in-test_torchinductor_opinfo.patch, PyTorch-2.9.0_fix-nccl-test-env.patch, PyTorch-2.9.0_fix-test_exclude_padding.patch, PyTorch-2.9.0_fix-test_version_error.patch, PyTorch-2.9.0_honor-XDG_CACHE_HOME.patch, PyTorch-2.9.0_increase-tolerance-in-test_transformers.patch, PyTorch-2.9.0_remove-faulty-close.patch, PyTorch-2.9.0_revert-pybind11-3-change.patch, PyTorch-2.9.0_skip-test_benchmark_on_non_zero_device.patch, PyTorch-2.9.0_skip-test_convolution1-on-H100.patch, PyTorch-2.9.0_skip-test_inductor_all_gather_into_tensor_coalesced.patch, PyTorch-2.9.0_skip-test_original_aten_preserved_pad_mm.patch, PyTorch-2.9.0_skip-test_override-without-CUDA.patch, PyTorch-2.9.0_skip-test_unbacked_reduction.patch, PyTorch-2.9.0_skip-tests-requiring-CUDA-12.8.patch, PyTorch-2.9.0_skip-unexpected-success-in-test_fake_export.patch, PyTorch-2.9.1_skip-RingFlexAttentionTest.patch

Thyre commented:
Test report by @Thyre
**FAILED**
Build succeeded for 3 out of 4 (total: 55 secs) (4 easyconfigs in total)
jrc0900.jureca - Linux Rocky Linux 9.6, AArch64, ARM UNKNOWN (neoverse_v2), 1 x NVIDIA NVIDIA GH200 480GB, 580.95.05, Python 3.9.21
See https://gist.github.com/Thyre/576f0dbeceb975733d860d97f16ca3fc for a full test report.

Flamefire committed:
Add patch for GCC 14 ARM builds

Flamefire commented:
Looks like I need to set those values earlier. Can you try again?

Flamefire commented:
> Failure may be caused by this: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=121027
> 
> There was a PR which should have worked around this, but seemingly the fix doesn't work? See also:
> 
>     * https://github.com/pytorch/pytorch/blob/f026b098e4319413db7d3fc1dbcb39dda69fcf0c/aten/src/ATen/native/cpu/Unfold2d.cpp#L172
> 
>     * [Build error: unrecognizable insn with using gcc-14 on aarch64 pytorch/pytorch#157842](https://github.com/pytorch/pytorch/issues/157842)


That is not included in this (or any) release yet. I'll add it to the patch list

> Maybe we need to patch `GCCcore/14.3.0` with this change? https://gcc.gnu.org/bugzilla/show_bug.cgi?id=121027#c9

Would be an option, not sure if it is worth it: This EC is included since EB 5.1.0

Flamefire committed:
Add workaround for GCC 14 ICE

Flamefire committed:
Skip tests requiring CUDA SM 9.0

Flamefire committed:
Add patch avoiding infinite test hang

Flamefire commented:
Test report by @Flamefire
**FAILED**
Build succeeded for 3 out of 4 (total: 23 hours 50 mins 38 secs) (4 easyconfigs in total)
n1702.barnard.hpc.tu-dresden.de - Linux RHEL 9.6, x86_64, Intel(R) Xeon(R) Platinum 8470 (sapphirerapids), Python 3.9.21
See https://gist.github.com/Flamefire/c44ac46441800b114fe0649766ca31d9 for a full test report.

Flamefire committed:
More patches

Flamefire committed:
Fix patched skip markers

Flamefire committed:
Add comment for DISABLE_ADDR2LINE

Flamefire commented:
Test report by @Flamefire
**FAILED**
Build succeeded for 3 out of 4 (total: 10 hours 30 mins 50 secs) (4 easyconfigs in total)
n1299.barnard.hpc.tu-dresden.de - Linux RHEL 9.6, x86_64, Intel(R) Xeon(R) Platinum 8470 (sapphirerapids), Python 3.9.21
See https://gist.github.com/Flamefire/d472b079fa275d681a1b9942cc2815df for a full test report.

Flamefire committed:
Set test timeout

Flamefire committed:
Add GCC 14 patch

Flamefire committed:
Add missing patch

Flamefire commented:
@Thyre 
> 	export/test_export_opinfo (431 failed, 201 passed, 40 skipped, 0 errors)
>	inductor/test_cpu_repro (4 failed, 211 passed, 526 skipped, 0 errors)
>	inductor/test_cpu_select_algorithm (58 failed, 31 passed, 1621 skipped, 0 errors)
>	inductor/test_cutlass_backend (8 failed, 142 passed, 2 skipped, 0 errors)
>	inductor/test_flex_attention (258 failed, 12 passed, 306 skipped, 0 errors)

I've seen issues with `test_flex_attention` including segfaults.
And @boegel has seen most of the failures in one report in `export/test_export_opinfo` too.

> Looks like a significant number of float32 tests failed on aarch64 / Neoverse V2. Looks like something like this PR might be related? https://github.com/pytorch/pytorch/pull/169937

Can you show a couple such failures, possibly grouped if they look the same?

Thyre commented:
I caught this from the snippet in the output log here: https://gist.github.com/Thyre/25e6d77bf7117f14c118bf0dd1a3e70f#file-pytorch-2-9-1-foss-2025b-cuda-12-9-1_partial-log-L449

I'll need to check if I still have the full log available. In the worst case, I have to re-run the build :see_no_evil:

Flamefire committed:
Add patches for test fixes and skip slow&disabled tests

Flamefire committed:
Add PyTorch-2.6.0_fix-server-in-test_control_plane

Flamefire committed:
Fix race condition in checking for disabled tests

Flamefire committed:
Remove pytest-shard

verdurin commented:
Worth noting here that I've been trying this on GH200 nodes, and the testing was taking over a week when the node died for other reasons.

Flamefire committed:
Add more patches

Flamefire commented:
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (total: 23 hours 44 mins 43 secs) (1 easyconfigs in total)
c51 - Linux Rocky Linux 9.6, x86_64, AMD EPYC 9334 32-Core Processor (zen4), 4 x NVIDIA NVIDIA H100, 580.65.06, Python 3.9.21
See https://gist.github.com/Flamefire/5f82b6abcf65a7e9d08b948be2baad10 for a full test report.

Flamefire commented:
Test report by @Flamefire
**FAILED**
Build succeeded for 0 out of 1 (total: 44 hours 28 mins 55 secs) (1 easyconfigs in total)
i8012 - Linux Rocky Linux 9.6, x86_64, AMD EPYC 7352 24-Core Processor (zen2), 8 x NVIDIA NVIDIA A100-SXM4-40GB, 580.65.06, Python 3.9.21
See https://gist.github.com/Flamefire/f6896e1e8697116ae98e2cd9f64f77f1 for a full test report.

Flamefire committed:
Fix using wrong OpenMP library

Flamefire committed:
Skip segfaulting flex_attention suite

Flamefire commented:
> Test report by @Flamefire
> **FAILED**
> Build succeeded for 0 out of 1 (total: 44 hours 28 mins 55 secs) (1 easyconfigs in total)
> i8012 - Linux Rocky Linux 9.6, x86_64, AMD EPYC 7352 24-Core Processor (zen2), 8 x NVIDIA NVIDIA A100-SXM4-40GB, 580.65.06, Python 3.9.21
> See https://gist.github.com/Flamefire/f6896e1e8697116ae98e2cd9f64f77f1 for a full test report.

There were 17 test failures.
>	distributed/algorithms/ddp_comm_hooks/test_ddp_hooks (5 failed, 1 passed, 0 skipped, 0 errors)

Fixed by PyTorch-2.9.1_fix-DDPCommHookType-python-3.13.patch

>	distributed/checkpoint/e2e/test_e2e_save_and_load (1 failed, 18 passed, 0 skipped, 0 errors)
>	distributed/checkpoint/test_checkpoint (2 failed, 6 passed, 0 skipped, 0 errors)
>	distributed/checkpoint/test_fsspec (1 failed, 2 passed, 0 skipped, 0 errors)

Issue with new Python version ("cannot pickle code object") which I'd just let fail

>	distributed/pipelining/test_stage (1 failed, 1 passed, 6 skipped, 0 errors)

Happens on nodes with > 4 GPUs: https://github.com/pytorch/pytorch/issues/162746

>	dynamo/test_error_messages (1 failed, 40 passed, 0 skipped, 0 errors)

pybind11 ABI changes while test checks for exact strings

>	functorch/test_ops (2 failed, 7489 passed, 2731 skipped, 0 errors)

Known failure but PyTorch-2.1.0_increase-tolerance-functorch-test_vmapvjpvjp.patch can't be easily updated

>	inductor/test_aot_inductor (4 failed, 418 passed, 445 skipped, 0 errors)

**IMPORTANT**: `cudaErrorInvalidDeviceFunction` depending on the order of `--cuda-compute-capabilities`, fixed by https://github.com/easybuilders/easybuild-easyblocks/pull/4092

> Could not count failed tests for the following test suites/files:
>	inductor/test_flex_attention (Undetected or did not run properly)

I tested this many times interactively and can't reproduce the segfaults nor find any hint for a cause, so I just skipped that.


So 9 out of 17 and the uncounted suite fixed, the remaining 8 should be ok to keep.

Flamefire committed:
Skip some tests failing on ARM

Flamefire committed:
Fix Python 3.13 compat

Flamefire commented:
> Issue with new Python version ("cannot pickle code object") which I'd just let fail

Fixed by using patch from upstream now

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 4 out of 4 (total: 37 hours 7 mins 12 secs) (1 easyconfigs in total)
node3907.accelgor.os - Linux RHEL 9.6, x86_64, AMD EPYC 7413 24-Core Processor (zen3), 1 x NVIDIA NVIDIA A100-SXM4-80GB, 590.48.01, Python 3.9.21
See https://gist.github.com/boegel/060a1ce806826b1d04f9b4e7a639fe9f for a full test report.

Flamefire commented:
After the benchmarks it seems like we should include ACL and MKL in our builds. Shall we merge this first and make new PRs for adding them or add them here (and in other open PyTorch PRs) before merging?

boegel commented:
> After the benchmarks it seems like we should include ACL and MKL in our builds. Shall we merge this first and make new PRs for adding them or add them here (and in other open PyTorch PRs) before merging?

I'd say we can do that in a separate PR, to make the change/fix stand out?

boegel commented:
@Flamefire please sync with `develop` now #24365 is merged...

boegel commented:
> > After the benchmarks it seems like we should include ACL and MKL in our builds. Shall we merge this first and make new PRs for adding them or add them here (and in other open PyTorch PRs) before merging?
> 
> I'd say we can do that in a separate PR, to make the change/fix stand out?

Also, it's not clear to me yet if it's only `imkl` or also [`oneDNN`](https://www.intel.com/content/www/us/en/developer/tools/oneapi/onednn.html) (see also https://github.com/uxlfoundation/oneDNN ?)

Just for context:
- see also https://github.com/easybuilders/easybuild/issues/931

Flamefire commented:
> I'd say we can do that in a separate PR, to make the change/fix stand out?

It was a change to yet unmerged ECs, hence the question. But of course can do in separate PRs after having those merged. Makes the changeset smaller.

> Also, it's not clear to me yet if it's only imkl or also [oneDNN](https://www.intel.com/content/www/us/en/developer/tools/oneapi/onednn.html) (see also [uxlfoundation/oneDNN](https://github.com/uxlfoundation/oneDNN?rgh-link-date=2026-03-30T13%3A54%3A23Z) ?)

`oneDNN` is included in PyTorch as a submodule and already used, but it is (sometimes) named MKL-DNN/MKLDNN which is the previous name: https://www.intel.com/content/www/us/en/docs/onednn/developer-guide-reference/2024-1/transitioning-from-intel-mkl-dnn-to-onednn.html

> In short, the migration can be as simple as just replacing all MKLDNN/mkldnn substrings with DNNL/dnnl.

 https://github.com/intel/mkl-dnn forwards to https://github.com/uxlfoundation/oneDNN

boegel commented:
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (total: 33 hours 22 mins 13 secs) (1 easyconfigs in total)
node4306.litleo.os - Linux RHEL 9.6, x86_64, AMD EPYC 9454P 48-Core Processor (zen4), 1 x NVIDIA NVIDIA H100 NVL, 580.95.05, Python 3.9.21
See https://gist.github.com/boegel/40208332798ff165baf4ad663c41f19f for a full test report.

boegel commented:
@boegelbot please test @ jsc-zen3-a100
CORE_CNT=16

Flamefire committed:
Disable sanity_check_pip_list

Flamefire committed:
Skip distributed/rpc/test_share_memory

Flamefire commented:
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (total: 9 hours 48 mins 12 secs) (1 easyconfigs in total)
n1089.barnard.hpc.tu-dresden.de - Linux RHEL 9.6, x86_64, Intel(R) Xeon(R) Platinum 8470 (sapphirerapids), Python 3.9.21
See https://gist.github.com/Flamefire/f04acc41c6dc5cbabc8bf1798f80883e for a full test report.

Flamefire commented:
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (total: 23 hours 55 mins 47 secs) (1 easyconfigs in total)
c14 - Linux Rocky Linux 9.6, x86_64, AMD EPYC 9334 32-Core Processor (zen4), 4 x NVIDIA NVIDIA H100, 580.159.04, Python 3.9.21
See https://gist.github.com/Flamefire/c74ac431f8cf5d09384d74f138f2be91 for a full test report.

Flamefire commented:
Test report by @Flamefire
**SUCCESS**
Build succeeded for 1 out of 1 (total: 30 hours 41 mins 14 secs) (1 easyconfigs in total)
i8004 - Linux Rocky Linux 9.6, x86_64, AMD EPYC 7352 24-Core Processor (zen2), 8 x NVIDIA NVIDIA A100-SXM4-40GB, 580.159.04, Python 3.9.21
See https://gist.github.com/Flamefire/3ab9b178655b5648a15f194157ce14f9 for a full test report.

VRehnberg commented:
I built and tested this on a GH200. Here are my notes from looking at the failed tests:

* distributed/optim/test_zero_redundancy_optimizer.py::TestZeroRedundancyOptimizerDistributed::test_add_param_group
* distributed/optim/test_zero_redundancy_optimizer.py::TestZeroRedundancyOptimizerDistributed::test_sharding
  * GPU-device out of range (only 1 GPU available but seems to try to use 2)
* dynamo/test_reconstruct.py::ReconstructTest::test_tma_stable_reconstruct
  * assertEqual which doesn't handle comparing tuples of TensorDescriptors, but the contents are in fact all equal
* inductor/test_cpu_repro.py::CPUReproTests::test_for_loop_collapsed
  * OpenMP pragma doesn't collapse for loop, seems fine otherwise. Also CPU so don't really care.
* inductor/test_cutlass_backend.py::TestCutlassBackend::test_standalone_runner
  * Unexpected non-zero exit code (SIGABRT?), unclear what this is about
* inductor/test_torchinductor_codegen_dynamic_shapes.py::DynamicShapesCodegenCpuTests::test_custom_op_fixed_layout_sequential_dynamic_shapes_cpu
  * fails to find \`;.\*ks.\*;\` but the code its search is python code with line breaks
* profiler/test_memory_profiler.py::TestMemoryProfilerE2E::test_memory_timeline
  * the ¿compiled c++ memory stack? is slightly different
* test_linalg.py::TestLinalgCPU::test_matmul_mv_cpu_float32
  * just over tolerance, just flaky it seems
* test_multiprocessing_spawn.py::ParallelForkServerPerfTest::test_forkserver_perf
  * Disabling with TORCH_MP_PARALLEL_START=0 seems to not work (start-up time is faster than expected).
* test_ops_jit.py::TestJitCPU::test_variant_consistency_jit_linalg_lstsq_cpu_complex64
  * unclear cause but also not something I expect anyone to use ever
* test_quantization.py::TestQuantizedOps::test_qsoftmax_qnnpack
  * Missing module `hypothesis`, is listed as "Install / Development extra requirements"
  * To be proper, should be added as a builddependency
  
  Oh and to be clear. I don't think any of these seem serious.

Flamefire commented:
@VRehnberg Thanks for testing! Some questions:
> GPU-device out of range (only 1 GPU available but seems to try to use 2)

Only the 2 tests? I see they are missing the checks, but others too. Where exactly does it error? Just to make sure I catch all cases

> Missing module hypothesis, is listed as "Install / Development extra requirements"
> To be proper, should be added as a builddependency

It is already: https://github.com/easybuilders/easybuild-easyconfigs/pull/24926/changes#diff-a646215821ed9006179cc29fe1e48b21b5d0b8d7803271e5942e58e21f62a4ccR219
Maybe your installation was damaged/ overwritten by a parallel one?

So only 11 failures? That should pass, especially if I can skip the ones requiring >=2 GPUs. Can you attach the log?

VRehnberg commented:
I lost the test logs. The above notes are from when I ran manually afterwards based on the summary. Though, now when I checked again a lot more tests failed (previously I ran until first failure(s) of the different files with reported failure and just assumed that would be the same as reported in the summary from the test step).

That is probably also why I lost hypothesis, so now I'm unsure what the actual error was.

I've started a rebuild with an `--upload-test-report`, hopefully a report with log should come in 25 hours.

VRehnberg commented:
@Flamefire here are the log from the entire build and the log for running the tests in particular:

- [easybuild-PyTorch-2.9.1-20260630.170055.vFOJz.log.gz](https://github.com/user-attachments/files/29586408/easybuild-PyTorch-2.9.1-20260630.170055.vFOJz.log.gz)
- [out.txt.gz](https://github.com/user-attachments/files/29586405/out.txt.gz)

VRehnberg commented:
And the test-reports in case those have something additional:

- [test-reports.tar.gz](https://github.com/user-attachments/files/29586615/test-reports.tar.gz)

Flamefire committed:
Fix test_linalg failures

Flamefire committed:
Fix tests on 1 GPU

Flamefire committed:
Fix test_memory_profiler on ARM

Flamefire commented:
@VRehnberg I went over the failures and added patches for most of them here and in https://github.com/easybuilders/easybuild-easyconfigs/pull/25736

Feel free to run the tests again maybe already on the one with ACL already.
Force-pushes due to cleanups and sync with the other PR and develop

VRehnberg commented:
I'll not be accessing the relevant cluster for the next four weeks, but I'll leave this unread and revisit when I'm back from vacation.

boegel commented:
@boegelbot please test @ jsc-zen3-a100
CORE_CNT=16

lexming commented:
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (total: 17 hours 49 mins 34 secs) (1 easyconfigs in total)
node804.hydra.os - Linux Rocky Linux 9.8 (Blue Onyx), x86_64, AMD EPYC 9275F 24-Core Processor (zen5), 1 x NVIDIA NVIDIA H200 NVL, 595.71.05, Python 3.9.25
See https://gist.github.com/lexming/d7d09701a0e79b5277ecc566e4160be9 for a full test report.

lexming reviewed (approved):
LGTM

Works well on older and newer x86/nvidia, so this seems ready to be merged.

VRehnberg commented:
> @VRehnberg I went over the failures and added patches for most of them here and in #25736
> 
> Feel free to run the tests again maybe already on the one with ACL already. Force-pushes due to cleanups and sync with the other PR and develop

Latest fixes gets me to less than 10 test failures as well on our GH200 system.

lexming commented:
@Flamefire merged, thanks!
